core/icon.ts

```diff
@@ -79,8 +79,11 @@ export abstract class Icon {
 
   /** Dispose of this icon. */
   dispose() {
-    dom.removeNode(this.iconGroup_);  // Dispose of and unlink the icon.
-    this.setVisible(false);           // Dispose of and unlink the bubble.
+    if (!this.getBlock().isDeadOrDying()) {
+      dom.removeNode(this.iconGroup_);
+    }
+    // TODO: Check if this is necessary.
```

Comment:
Nit: associate todo with an issue

Reply:
It is necessary because the bubble lives on the bubble canvas, and is not an SVG element of the block. It will not get automatically garbage collected, so we need to dispose of it separately.